### PR TITLE
FALCON-2235 Suspend/Resume API support for extension job (user extension)

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 1]
```diff
@@ -221,52 +221,81 @@ public APIResult schedule(@PathParam("job-name") String jobName,
...
-                if (entity.getEntityType().isSchedulable()) {
-                    if (getWorkflowEngine(entity).isActive(entity)) {
-                        getWorkflowEngine(entity).suspend(entity);
-                    }
-                }
-            }
-        } catch (FalconException | IOException e) {
-            LOG.error("Error when scheduling extension job: " + jobName + ": ", e);
+        try {
+            suspendEntities(extensionJobsBean, coloExpr, request);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Why not use` entityMap = getJobEntities(extensionJobsBean)` as it is used in the schedule method?

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 2]
```diff
@@ -221,52 +221,81 @@ public APIResult schedule(@PathParam("job-name") String jobName,
...
         }
         return new APIResult(APIResult.Status.SUCCEEDED, "Extension job " + jobName + " suspended successfully");
     }
 
+    private void suspendEntities(ExtensionJobsBean extensionJobsBean, String coloExpr, final HttpServletRequest request)
+        throws FalconException {
+        List<String> processes = extensionJobsBean.getProcesses();
+        List<String> feeds = extensionJobsBean.getFeeds();
+        final HttpServletRequest bufferedRequest = new BufferedRequest(request);
+        suspendEntities(coloExpr, feeds, EntityType.FEED.name(), bufferedRequest);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Don't see a need for existence of 2 suspendEntities method and invoke it twice, if we followed the same code logic used in schedule.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 3]
```diff
@@ -221,52 +221,81 @@ public APIResult schedule(@PathParam("job-name") String jobName,
...
+        suspendEntities(coloExpr, processes, EntityType.PROCESS.name(), bufferedRequest);
+    }
+
+    private void resumeEntities(ExtensionJobsBean extensionJobsBean, String coloExpr, final HttpServletRequest request)
+        throws FalconException {
+        List<String> processes = extensionJobsBean.getProcesses();
+        List<String> feeds = extensionJobsBean.getFeeds();
+        final HttpServletRequest bufferedRequest = new BufferedRequest(request);
+        resumeEntities(coloExpr, feeds, EntityType.FEED.name(), bufferedRequest);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Don't see a need for existence of 2 resumeEntities method and invoke it twice, if we followed the same code logic used in schedule.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 4]
```diff
@@ -221,52 +221,81 @@ public APIResult schedule(@PathParam("job-name") String jobName,
...
-            for (Entity entity : entities) {
-                if (entity.getEntityType().isSchedulable()) {
-                    if (getWorkflowEngine(entity).isSuspended(entity)) {
-                        getWorkflowEngine(entity).resume(entity);
-                    }
-                }
-            }
-        } catch (FalconException | IOException e) {
-            LOG.error("Error when resuming extension job " + jobName + ": ", e);
+            resumeEntities(extensionJobsBean, coloExpr, request);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: As comment that was made for suspend. Why not use getJobEntities?

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 1 above)
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Ack. Made the changes.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 2 above)
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Ack. Made the changes.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 3 above)
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Ack. Made the changes.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 4 above)
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Ack. Made the changes.

File: unit/src/main/java/org/apache/falcon/unit/LocalExtensionManager.java [hunk 5]
```diff
@@ -39,7 +39,7 @@
 /**
  * A proxy implementation of the extension operations in local mode.
  */
-public class LocalExtensionManager extends AbstractExtensionManager {
+class LocalExtensionManager extends AbstractExtensionManager {
```
Comment (pallavi-rao) [inline unit/src/main/java/org/apache/falcon/unit/LocalExtensionManager.java]: How about using getJobEntities even for suspend and resume?

**Pull request closed without merging**